### PR TITLE
Added support for thenAnswer like the java version of Mockito.

### PR DESCRIPTION
**Pull request by stubbsd**
Example:

```pathon
m = mock()
when(m).do_times(any(), any()).thenAnswer(lambda one, two: one * two)
self.assertEquals(20, m.do_times(5, 4))
```

### CODE REVIEW

**Comment by kaste** [comment]
Yeah, that's probably a good feature :+1:

File: mockito/invocation.py [hunk 1]
```diff
@@ -177,14 +178,18 @@ def thenRaise(self, *exceptions):
             self.__then(Raise(exception))
         return self
 
+    def thenAnswer(self, *callables):
+        for callable in callables:
+            self._AnswerSelector__then(ReturnAnswer(self.invocation.mock, callable))
```
Comment (kaste) [inline mockito/invocation.py]: The `_AnswerSelector__then` python extravaganza shouldn't be necessary. `__then` should just work.

Comment (stubbsd) [inline mockito/invocation.py]: Yes, your are absolutely correct. 

I pulled the code out from a project that I'd jerry-rigged to work without change the library.

File: mockito/invocation.py [hunk 2]
```diff
@@ -215,3 +220,15 @@ def __init__(self, return_value):
...
+    def __init__(self, mock, answerable):
+        self.answerable = answerable
+        self.mock = mock
+
+    def answer(self):
+        if len(inspect.getargspec(self.answerable).args) == 0:
+            return self.answerable()
+        else:
+            return self.answerable(*self.mock.invocations[0].params)
+
```
Comment (kaste) [inline mockito/invocation.py]: The `if...then` is not needed b/c if params is empty (the initial value is an empty tuple I think. The `*params` would still work. Support for `kwargs` here would be :sunglasses:

Comment (stubbsd) [inline mockito/invocation.py]: Sadly @kaste my Python-Fu is lacking.

Updated with your suggestions.

File: mockito/invocation.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 #  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #  THE SOFTWARE.
+import inspect
```
Comment (kaste) [inline mockito/invocation.py]: Now unused

File: mockito/invocation.py [hunk 4]
```diff
@@ -215,3 +220,13 @@ def __init__(self, return_value):
...
     def answer(self):
         return self.return_value
+
+class ReturnAnswer(object):
+    def __init__(self, mock, answerable):
+        self.answerable = answerable
+        self.mock = mock
+
+    def answer(self):
+        print dir(self.mock.invocations[0])
```
Comment (kaste) [inline mockito/invocation.py]: Oops left over. I should probably teach Travis to run a linter.

File: mockito/invocation.py [hunk 5]
```diff
@@ -215,3 +220,13 @@ def __init__(self, return_value):
...
+        return self.answerable(*self.mock.invocations[0].params, **self.mock.invocations[0].named_params)
```
Comment (kaste) [inline mockito/invocation.py]: Probably ‘current_invocation = ...’

**Comment by kaste** [comment]
LGTM! :+1: 

Pushed as v0.6.1

**Pull request merged**